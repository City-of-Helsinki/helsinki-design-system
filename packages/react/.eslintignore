**/*.d.ts
webpack.config.js
.templates
lib/
**/*.stories.tsx
