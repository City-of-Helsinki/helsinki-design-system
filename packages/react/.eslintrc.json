{
  "parser": "@typescript-eslint/parser",
  "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/eslint-recommended",
    "plugin:react/recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:import/errors",
    "plugin:import/warnings",
    "plugin:import/typescript",
    "plugin:jest/recommended",
    "airbnb",
    "prettier/@typescript-eslint",
    "plugin:prettier/recommended" // this should always be the last item on this array
  ],
  "plugins": ["react", "@typescript-eslint", "prettier", "jest", "import"],
  "parserOptions": {
    "ecmaVersion": 2018,
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "ignorePatterns": ["*.module.css"],
  "rules": {
    "no-use-before-define": "off",
    "@typescript-eslint/no-use-before-define": ["error"],
    "import/no-extraneous-dependencies": [
      "error",
      {
        "devDependencies": ["**/*.test.ts", "**/*.test.tsx", "**/test*.ts*", "**/*.stories.tsx"]
      }
    ],
    "import/order": [
      "error",
      {
        "groups": ["builtin", "external", "internal", ["parent", "sibling", "index"]],
        "newlines-between": "always"
      }
    ],
    "react/jsx-filename-extension": "off",
    "react/jsx-one-expression-per-line": "off",
    "react/jsx-props-no-spreading": "off",
    "react/jsx-curly-newline": "off",
    "@typescript-eslint/explicit-function-return-type": "off",
    "react/prop-types": "off",
    "import/extensions": "off",
    "import/prefer-default-export": "off",
    "jsx-a11y/label-has-associated-control": [
      2,
      {
        "components": ["Label"],
        "required": {
          "some": ["nesting", "id"]
        },
        "allowChildren": false
      }
    ],
    "@typescript-eslint/explicit-member-accessibility": "off",
    "react/state-in-constructor": [2, "never"],
    "no-unused-expressions": [2, {
      "allowTernary": true
    }],
    "no-unused-vars": "off",
    "@typescript-eslint/no-unused-vars": ["error"]
  },
  "settings": {
    "react": {
      "version": "detect"
    },
    "import/resolver": {
      "node": {
        "extensions": [".js", ".jsx", ".ts", ".tsx"],
        "paths": ["src"]
      }
    }
  },
  "env": {
    "browser": true,
    "jest": true
  },
  "globals": {
    "JSX": true
  }
}
