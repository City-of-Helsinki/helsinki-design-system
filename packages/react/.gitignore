node_modules/
storybook-static/
lib/
_bundles/
*.css.d.ts
.loki/current
.loki/difference

*storybook.log
