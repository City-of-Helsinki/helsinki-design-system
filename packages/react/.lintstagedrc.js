module.exports = {
  'src/**/*.{ts,tsx}': 'eslint',
}
