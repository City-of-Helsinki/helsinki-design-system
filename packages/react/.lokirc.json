{
  "chromeRetries": 2,
  "configurations": {
    "chrome.laptop": {
      "target": "chrome.docker",
      "width": 1366,
      "height": 768,
      "mobile": false
    },
    "chrome.iphone7": {
      "target": "chrome.docker",
      "preset": "iPhone 7"
    }
  },
  "skipStories": "Icons/.*|Button/*|Tag/*|Hero/*|Koros/*|Checkbox/*",
  "pixelmatch": {
    "threshold": 0.2,
    "alpha": 1,
    "diffColor": [255, 0, 255]
  }
}
