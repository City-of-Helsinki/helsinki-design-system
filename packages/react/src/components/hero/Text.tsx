import React from 'react';

import styles from './Hero.module.scss';
import classNames from '../../utils/classNames';
import { AllElementPropsWithoutRef } from '../../utils/elementTypings';

export const Text = (props: AllElementPropsWithoutRef<'p'>): React.ReactElement => {
  const { className, children, ...elementProps } = props;
  const classList = classNames(styles.text, className);
  return (
    <p {...elementProps} className={classList}>
      {children}
    </p>
  );
};
