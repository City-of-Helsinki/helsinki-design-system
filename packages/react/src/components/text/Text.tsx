import React from 'react';

import styles from './text.module.scss';

export interface TextProps {
  color?: 'standard' | 'brand' | 'critical' | 'secondary' | 'info';
  as?: 'span' | 'em' | 'strong' | 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
  size?: 'xs' | 's' | 'm' | 'l' | 'xl' | 'xxl' | 'xxl' | 'xxxl';
  bold?: boolean;
  italic?: boolean;
}

const Text: React.SFC<TextProps> = ({ color = 'standard', size, as: Element = 'span', bold, italic, children }) => {
  return (
    <Element
      className={`${styles.text} ${styles[color]} ${styles[Element]} ${styles[size]} ${bold && styles.bold} ${italic &&
        styles.italic}`}
    >
      {children}
    </Element>
  );
};

export default Text;
