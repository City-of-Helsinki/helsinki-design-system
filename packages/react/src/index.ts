export * from './icons';
export * from './components';
export * from './ssr';
