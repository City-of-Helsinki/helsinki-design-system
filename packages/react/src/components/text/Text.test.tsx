import React from 'react';
import { render } from '@testing-library/react';

import Text from './Text';

describe('<Text /> spec', () => {
  it('renders the component', () => {
    const { asFragment } = render(<Text />);
    expect(asFragment()).toMatchSnapshot();
  });
});
