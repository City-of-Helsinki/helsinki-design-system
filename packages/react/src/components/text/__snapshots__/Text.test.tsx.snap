// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> spec renders the component 1`] = `
<DocumentFragment>
  <span
    class="text standard span undefined undefined undefined"
  />
</DocumentFragment>
`;
